runtime/compiler/trj9/compile/J9Compilation.hpp

```diff
@@ -284,6 +284,15 @@ class OMR_EXTENSIBLE Compilation : public OMR::CompilationConnector
    //
    bool supportsQuadOptimization();
 
+   // Debug Counter Relocation
+   typedef std::pair<const char *, const char *> DebugCounterNameEntry;
+   typedef TR::typed_allocator<DebugCounterNameEntry, TR::Allocator> DebugCounterNameMapAllocator;
+   typedef std::map<const char *, const char *, std::less<const char *>, DebugCounterNameMapAllocator> DebugCounterNameMap;
+   void mapKeyToCounterName(const char *key, const char *counterName);
+   const char *getCounterName(const char *key);
+   uint32_t calculateSizeOfCounterNames(uint32_t &numDebugCounterNames);
+   DebugCounterNameMap &getDebugCounterNameMap() { return _debugCounterNameMap; }
```

Comment:
These new APIs should have Doxygen documentation, unless this is an extensible class and they've been already documented in OMR. Similarly for the new field added as well.